### PR TITLE
enable using mouse on the radial menu

### PR DESCRIPTION
**Pull request by rafccq**
Solves #107

### CODE REVIEW

**Comment by fgsfdsfgs** [comment]
What's the whole current device thing for? Is it not enough to just check if `dx`/`dy` are non-zero, like the rest of the input code does?

**Comment by cylonicboom** [comment]
I'd really like to be able to play with a left-analog stick and mouse for aiming. This would explicitly break that.

When the check is removed, you can't move the radial dial with the xbox sticks after using the mouse unless you close and re-open the radial menu.

**Comment by cylonicboom** [comment]
> I'd really like to be able to play with a left-analog stick and mouse for aiming. This would explicitly break that.
> 
> When the check is removed, you can't move the radial dial with the xbox sticks after using the mouse unless you close and re-open the radial menu.

Also it's not just this one case: it's preferable to map the steam deck gyro / stick / trackpads to the mouse, which would cause unfriendly behavior unless key presses are being handled correctly in the game code.

**Comment by fgsfdsfgs** [comment]
That can be solved by just adding up movement from the sticks and the mouse instead of making them exclusive. Let me see what I can do.

**Comment by rafccq** [comment]
@fgsfdsfgs I didn't use dx/dy to keep the slot selected, and not return to the center if there was no movement.
I added the device thing so the game detects the last device you used, and use only that one for the radial menu input, but
yes sounds better to just add the movements of mouse and sticks, I didn't think of it. 
That will make it work with control schemes such as the ones mentioned by @NeonNyan

**Comment by fgsfdsfgs** [comment]
Something like this would probably work fine, though it does need some tuning.
[radial-mouse.patch](https://github.com/fgsfdsfgs/perfect_dark/files/12453099/radial-mouse.patch)

**Comment by rafccq** [comment]
Thanks, I'll test it out and tune if needed.

**Comment by AXDOOMER** [comment]
The new code doesn't use the same indentation as the rest of the file. This should be fixed.

File: port/src/input.c [hunk 1]
```diff
@@ -479,7 +480,7 @@ static inline s32 inputAxisScale(s32 x, const s32 deadzone, const f32 scale)
 	if (abs(x) < deadzone) {
 		return 0;
 	} else {
-		// rescale to fit the non-deadzone range
+	 	// rescale to fit the non-deadzone range
```
Comment (AXDOOMER) [inline port/src/input.c]: This new random whitespace should be removed.

File: src/game/activemenutick.c [hunk 2]
```diff
@@ -63,6 +67,24 @@ void amTick(void)
...
+                        am->mousey += mdy * 4.f;
+                        am->mousex = (am->mousex > 127.f) ? 127.f : (am->mousex < -128.f) ? -128.f : am->mousex;
+                        am->mousey = (am->mousey > 127.f) ? 127.f : (am->mousey < -128.f) ? -128.f : am->mousey;
+                    }
+                    const s32 newstickx = (s32)cstickx + (s32)am->mousex;
+                    const s32 newsticky = (s32)csticky - (s32)am->mousey;
+                    cstickx = (newstickx < -128) ? -128 : (newstickx > 127) ? 127 : newstickx;
+                    csticky = (newsticky < -128) ? -128 : (newsticky > 127) ? 127 : newsticky;
+                }
+#endif
```
Comment (AXDOOMER) [inline src/game/activemenutick.c:86]: The new code doesn't use the same indentation as the rest of the file. This should be fixed.

File: src/game/activemenutick.c [hunk 3]
```diff
@@ -337,6 +359,12 @@ void amTick(void)
 				}
 			}
 		}
+#ifndef PLATFORM_N64
+        else {
+            g_AmMenus[g_AmIndex].mousex = 0.f;
+            g_AmMenus[g_AmIndex].mousey = 0.f;
+        }
+#endif
```
Comment (AXDOOMER) [inline src/game/activemenutick.c:367]: This code is also not indented the same as the rest of the code. This file uses tabs put your editor is inserting spaces.

File: src/include/types.h [hunk 4]
```diff
@@ -4306,6 +4306,11 @@ struct activemenu {
 	/*0x32*/ u8 allbots; // when player holds R on the bot command screen
 	/*0x33*/ u8 prevallbots; // used when opening "Pick Target" menu for attack command
 	/*0x34*/ s8 origscreennum; // original screen number before using allbots
+
+#ifndef PLATFORM_N64
+    /*    */ f32 mousex;
+    /*    */ f32 mousey;
+#endif
```
Comment (AXDOOMER) [inline src/include/types.h:4310]: Also not indented as the rest.

**Review by AXDOOMER — CHANGES_REQUESTED** [review]

**Comment by fgsfdsfgs** [comment]
The whitespace issues are because you're using spaces instead of tabs, unlike the rest of the code.

**Pull request merged**